sdk/tests/iot/hub/test_az_iot_hub_client_twin.c

```diff
@@ -478,6 +518,8 @@ int test_az_iot_hub_client_twin()
         test_az_iot_hub_client_twin_parse_received_topic_reported_props_version_found_succeed),
     cmocka_unit_test(
         test_az_iot_hub_client_twin_parse_received_topic_reported_props_no_version_found_succeed),
+    cmocka_unit_test(test_az_iot_hub_client_twin_parse_received_topic_bad_request_succeed),
```

Comment:
Update the names if taking up the suggested changes.

Reply:
will do thanks.

Reply:
After thinking about it some more, I think the "success/succeed" should stay for the tests that pass with AZ_OK.  This result means a valid topic was successfully parsed, which is true in the case of the 400+ result codes.  To minimize confusion, I will change the names of those tests to use the result code number, avoiding the head scratching "bad_request_succeed" for example.